### PR TITLE
Add new image for bridge server e2e tests

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** 
Creates and builds an e2e test image for an OpAMP server with endpoints available for e2e testing

**Link to tracking Issue(s):**

- Resolves: #3094

**Testing:** This is testing

**Documentation:** n/a

### CODE REVIEW

File: .github/workflows/publish-autoinstrumentation-e2e-images.yaml
Comment (jaronoff97) [inline .github/workflows/publish-autoinstrumentation-e2e-images.yaml:1]: this was renamed!

File: tests/test-e2e-apps/bridge-server/data/agent.go
Comment (jaronoff97) [inline tests/test-e2e-apps/bridge-server/data/agent.go:1]: much of this code was copied from [here](https://github.com/open-telemetry/opamp-go/blob/ed38d5f4bf930b57e04581919fbfb676aaa5a5af/internal/examples/server/)

**Comment by jaronoff97** [comment]
a follow up will use the published test image!

**Review by pavolloffay — APPROVED** [review]

**Review by swiatekm — APPROVED** [review]
This is a fair amount of code, so I think we should set up some basic tooling for it. We should run golangci-lint on it, for one. I'm ok with that happening in a follow-up though.

**Pull request merged**